File: BackEnd/springboot.ecommerce/src/main/java/com/swayampp/ecommerce/entity/OrderItem.java
package com.swayampp.ecommerce.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name="order_items")
@Getter
@Setter
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "product_id")
    private int productId;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "unit_price")
    private float unitPrice;

    @Column(name = "quantity")
    private int quantity;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "order_id")
    private Order order;

    @Override
    public String toString() {
        return "OrderItem{" +
                "id=" + id +
                ", productId=" + productId +
                ", imageUrl='" + imageUrl + '\'' +
                ", unitPrice=" + unitPrice +
                ", quantity=" + quantity +
                '}';
    }
}

File: FrontEnd/angular-ecommerce/src/app/common/order-item.ts
import { CartItem } from "./cart-item"

export class OrderItem {
    public productId: number;
    public imageUrl: string;
    public unitPrice: number;
    public quantity: number;

    constructor(cartItem: CartItem) {
        this.productId = cartItem.product.id;
        this.imageUrl = cartItem.product.imageUrl;
        this.unitPrice = cartItem.product.unitPrice;
        this.quantity = cartItem.quantity;
    }
}
